tests/functional/src/main/java/org/mule/functional/junit4/FunctionalTestCase.java

```diff
@@ -81,6 +82,9 @@ public FunctionalTestCase() {
   public SystemProperty muleDisableAsyncRetryPolicyOnSourcesProperty =
       new SystemProperty(DISABLE_ASYNC_RETRY_POLICY_ON_SOURCES, "true");
 
+  @Rule
+  public SystemProperty emitNotifications = new SystemProperty(EMIT_POLLING_SOURCE_NOTIFICATIONS, "true");
```

Comment:
only enable this for the test case that requires it.

Reply:
this is a problem because the field in the PollingSourceWrapper is static:
`  private static final boolean EMIT_NOTIFICATIONS = parseBoolean(getProperty(EMIT_POLLING_SOURCE_NOTIFICATIONS));
`
so when other test suites run (that don't have the @Rule set) they initialize the PollingSourceWrapper class and set the static field to false. I tried to access the PollingSourceWrapper class through reflection to change it before the tests, but was unable to do so because of class loading isolation. This was the way I found to make the tests possible.

Reply:
then simply don't make it static and use a Rule in the tests that you want this enabled on

Reply:
was not static, changed it because of this https://github.com/mulesoft/mule/pull/11666#discussion_r951466616 :)

Reply:
I understand that, but if the consequence is having to to enable this for the entire suite then it's a no-go. Another alternative is to have it as static but mutable through a setter. You can then enable it in your test and disable it afterwards